### PR TITLE
app template

### PR DESCRIPTION
**Pull request by mikelpmc**

### CODE REVIEW

File: staff/mikelpmc/js/challenges/camel-case/scripts/camel-case.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+const toCamelCase = str => {
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/camel-case/scripts/camel-case.js]: TIP 'use strict';

Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/camel-case/scripts/camel-case.js]: WARN ```arrow functions```are not allowed. this is ES5, not 6.  use a function instead.

File: staff/mikelpmc/js/challenges/camel-case/scripts/camel-case.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    if (typeof str !== 'string') throw Error('input str is not a string');
+
+    const res = str
+        .toLowerCase()
+        .split(' ')
+        .map((r, i, arr) => r[0].toUpperCase() + r.substr(1))
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/camel-case/scripts/camel-case.js]: WARN ```arrow functions```are not allowed. this is ES5, not 6.  use a function instead.

File: staff/mikelpmc/js/challenges/camel-case/scripts/main.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
+const res = toCamelCase('Hello My World');
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/camel-case/scripts/main.js]: TIP 'use strict';

File: staff/mikelpmc/js/challenges/count-chars/scripts/count-chars.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+    return str.length;
+};
+
+const filterLetter = (str, letter) => {
+    if (typeof str !== 'string') throw Error('input str is not a string');
+
+    return str
+        .split('')
+        .filter(s => s === letter)
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/count-chars/scripts/count-chars.js]: WARN ```arrow functions```are not allowed. this is ES5, not 6.  use a function instead.

File: staff/mikelpmc/js/challenges/count-chars/scripts/count-chars.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+// Funcion para contar los caracteres
+
+const countChars = str => {
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/count-chars/scripts/count-chars.js]: WARN ```arrow functions```are not allowed. this is ES5, not 6.  use a function instead.

File: staff/mikelpmc/js/challenges/count-chars/scripts/count-chars.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+// Funcion para contar los caracteres
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/count-chars/scripts/count-chars.js]: TIP 'use strict';

File: staff/mikelpmc/js/challenges/count-chars/scripts/main.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
+let count = countChars('hello world');
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/count-chars/scripts/main.js]: WARN ```let```is not allowed. this is ES5, not 6. use ```var``` instead.

File: staff/mikelpmc/js/challenges/count-chars/scripts/main.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+console.log(
+    'countChars("hello world") should return 11: ',
+    count === 11,
+    count
+);
+
+try {
+    let count = countChars(true);
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/count-chars/scripts/main.js]: WARN ```let```is not allowed. this is ES5, not 6. use ```var``` instead.

File: staff/mikelpmc/js/challenges/count-chars/scripts/main.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+} catch (error) {
+    console.log(
+        'countChars(true) should throw an error',
+        error !== undefined,
+        error
+    );
+}
+
+try {
+    let count = countChars(1);
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/count-chars/scripts/main.js]: WARN ```let```is not allowed. this is ES5, not 6. use ```var``` instead.

File: staff/mikelpmc/js/challenges/count-chars/scripts/main.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+} catch (err) {
+    console.log('countChars(1) should throw an error', err !== undefined, err);
+}
+
+try {
+    let count = countChars([]);
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/count-chars/scripts/main.js]: WARN ```let```is not allowed. this is ES5, not 6. use ```var``` instead.

File: staff/mikelpmc/js/challenges/count-chars/scripts/main.js [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+} catch (err) {
+    console.log('countChars([]) should throw an error', err !== undefined, err);
+}
+
+try {
+    let count = countChars('abracadabra', '...');
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/count-chars/scripts/main.js]: WARN ```let```is not allowed. this is ES5, not 6. use ```var``` instead.

File: staff/mikelpmc/js/challenges/count-chars/scripts/main.js (hunk 7 above)
Comment (manuelbarzi) [reply staff/mikelpmc/js/challenges/count-chars/scripts/main.js]: TIP 'use strict';

File: staff/mikelpmc/js/challenges/count-words/scripts/count-words.js [hunk 12]
```diff
@@ -0,0 +1,3 @@
+const countWords = str => {
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/count-words/scripts/count-words.js]: TIP 'use strict';

File: staff/mikelpmc/js/challenges/count-words/scripts/main.js [hunk 13]
```diff
@@ -0,0 +1,8 @@
+const str = 'Mikel Parra';
```
Comment (manuelbarzi) [inline staff/mikelpmc/js/challenges/count-words/scripts/main.js]: TIP 'use strict';

**Comment by mikelpmc** [comment]
:(

**Review by agandia9 — COMMENTED** [review]
nozing more to say!

File: .vscode/launch.json [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "type": "node",
+            "request": "launch",
+            "name": "Launch Program",
+            "program": "${file}"
+        }
+    ]
+}
```
Comment (MinaZhen) [inline .vscode/launch.json:14]: Que hace un jota-son por aquí?? 🤣

**Review by MinaZhen — APPROVED** [review]
+3!

File: .vscode/launch.json (hunk 14 above)
Comment (mikelpmc) [reply .vscode/launch.json:14]: Cositas del visual studio 🤕

File: staff/mikelpmc/js/challenges/scripts/camelCase.js [hunk 15]
```diff
@@ -0,0 +1,11 @@
+function toCamelCase(str) {
```
Comment (agandia9) [inline staff/mikelpmc/js/challenges/scripts/camelCase.js:3]: use strict!

File: staff/mikelpmc/js/challenges/scripts/randomCase.js [hunk 16]
```diff
@@ -0,0 +1,16 @@
+function toRandomCase(str) {
```
Comment (agandia9) [inline staff/mikelpmc/js/challenges/scripts/randomCase.js:3]: use strict!

File: staff/mikelpmc/js/challenges/scripts/randomCase.test.js [hunk 17]
```diff
@@ -0,0 +1,22 @@
+var input = 'hola que tal';
```
Comment (agandia9) [inline staff/mikelpmc/js/challenges/scripts/randomCase.test.js:3]: use strict!

Comment (mikelpmc) [reply staff/mikelpmc/js/challenges/scripts/randomCase.test.js:3]: done

**Pull request merged**